kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -79,6 +82,118 @@ public static void validateSchema(StructType schema, boolean isColumnMappingEnab
     validateSupportedType(schema);
   }
 
+  /**
+   * Performs the following validations on an updated table schema:
+   *
+   * <p>The following checks are performed:
+   *
+   * <ul>
+   *   <li>No duplicate columns are allowed
+   *   <li>Column names contain only valid characters
+   *   <li>Data types are supported
+   *   <li>No new non-nullable fields are added
+   *   <li>Physical column name consistency is preserved in the new schema
+   *   <li>Partition columns are not modified
+   *   <li>Nested IDs for array/map types are preserved in the new schema
+   *   <li>No type changes
+   * </ul>
+   */
+  public static void validateUpdatedSchema(
+      StructType currentSchema,
+      StructType newSchema,
+      List<String> partitionCols,
+      Metadata metadata) {
+    validateSchema(newSchema, true /*columnMappingEnabled*/);
+    Map<String, StructField> currentColumnPathToField = columnPathToField(currentSchema);
+    Map<String, StructField> updatedColumnPathToField = columnPathToField(newSchema);
+    ColumnMapping.validateColumnIds(
+        currentColumnPathToField,
+        updatedColumnPathToField,
+        TableConfig.ICEBERG_COMPAT_V2_ENABLED.fromMetadata(metadata));
+    validateUpdatedSchemaCompatibility(
+        newSchema, currentColumnPathToField, updatedColumnPathToField, partitionCols);
+  }
+
+  /* Verifies that no non-nullable fields are added, no type changes are performed and that
+  partition columns are not modified */
+  private static void validateUpdatedSchemaCompatibility(
+      StructType newSchema,
+      Map<String, StructField> currentColumnPathToStruct,
+      Map<String, StructField> updatedColumnPathToStruct,
+      List<String> partitionCols) {
+    validatePartitionColumns(newSchema, partitionCols);
```

Comment:
Can this be moved to `validateUpdatedSchema` ? I'd prefer our method calls be as flat as possible. i.e. instead of having some long chain of methodA -> methodB -> methodC, methodD ... if we can flatten and push methods down it makes it a bit easier to discern the total validation methods invoked

Reply:
done